### DISCUSSION TITLE
Github Copilot Chat having errors "Details: RuntimeError: Function documentContext.Selections execution failed."

### DISCUSSION THREAD

**Discussion by AlanParadis** [community/community#55817] [Copilot Conversations]
### Select Topic Area

Bug

### Body

When I ask Github Copilot Chat about codebase and architecture this error occur and it is not able to answer my questions.

I have a project with a Widget, a Checkbox and a BitmapCheckbox class all in their own files. 

My conversation :
user : "Explain how the BitmapCheckBox heritate from CheckBox wich heritate from Widget. Describe the architecture"
copilot : "Sorry, error fulfilling request. Details: RuntimeError: Function documentContext.Selections execution failed. System.ArgumentOutOfRangeException: Index was out of range. Must be non-negative and less than the size of the collection. (Parameter 'startIndex')"

Visual Studio Community Version : Version 17.6.0
Github Copilot Version : Version 0.1.651.34310

Let me know if there is any logs that I can share for furthermore debugging

**Comment by Akash1134** [comment]
Hi @AlanParadis ,

Are you using Copilot Chat with the VSCode insiders edition? More details [here](https://code.visualstudio.com/blogs/2023/03/30/vscode-copilot). Thanks!

**Reply by AlanParadis** [reply]
I'm using [Copilot Preview](https://marketplace.visualstudio.com/items?itemName=VisualStudioExptTeam.VSGitHubCopilot) extension the Visual Studio Comunity 2022 not Visual Studio Code

**Reply by Vana12330** [reply]
,😉

**Comment by Marvvyn** [comment]
It does not work at all in VS 17.6 Enterprise. I get these errors:

[Conversations Information] Activated Conversations version 0.1.651-beta+8606258f14.
[Conversations Critical] Exception during package initialization: Microsoft.VisualStudio.Composition.CompositionFailedException: Expected 1 export(s) with contract name "Microsoft.VisualStudio.LanguageServices.VisualStudioWorkspace" but found 0 after applying applicable constraints.
   at Microsoft.VisualStudio.Composition.ExportProvider.GetExports(ImportDefinition importDefinition)
   at Microsoft.VisualStudio.Composition.ExportProvider.GetExports[T,TMetadataView](String contractName, ImportCardinality cardinality)
   at Microsoft.VisualStudio.Composition.ExportProvider.GetExport[T,TMetadataView](String contractName)
   at Microsoft.VisualStudio.Composition.ExportProvider.GetExportedValue[T]()
   at Microsoft.VisualStudio.ComponentModelHost.ComponentModel.GetService[T]()
   at Microsoft.VisualStudio.Conversations.Utils.VsWorkspaceService.<InitializeAsync>d__10.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.VisualStudio.Conversations.ConversationsPackage.<>c__DisplayClass12_0.<<InitializeAsync>b__2>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at Microsoft.VisualStudio.Conversations.ConversationsPackage.<InitializeAsync>d__12.MoveNext()

**Reply by tscho5590** [reply]
Same. Any update?
![image](https://github.com/community/community/assets/13469656/cd5fd04d-9d48-4514-ab99-98fe976caee3)
![image](https://github.com/community/community/assets/13469656/e122485f-e63d-4931-9be7-ee0101c4e161)

**Reply by redemptions** [reply]
Also having this problem in Visual Studio 2022 Community, but NOT VSCode Insiders

**Reply by NikolayIT** [reply]
I am also having this problem in `Microsoft Visual Studio Enterprise 2022 (64-bit) - Current` version `17.6.2`.
The issue occurs when I simply open Visual Studio and try to use Copilot chat without selecting any code in the current document.

**Reply by Kesendo** [reply]
Same problem on a simple test with a checkbox question too

me; the function Form1_Load should make a checkbox read only
copilot: private void Form1_Load(object sender, EventArgs e)
{
    checkBox1.Enabled = false;
}
me: the checkbox is now disabled, and the layout changed, I am looking for something the value can't changed
copilot: Sorry, error fulfilling request. Details: RuntimeError: Function documentContext.Selections execution failed. System.ArgumentOutOfRangeException: Length cannot be less than zero. (Parameter 'length')

Microsoft Visual Studio Community 2022 (64-Bit) - Current
Version 17.6.0

**Reply by slFlak** [reply]
Which framework do you guys use in the projects your testing with? 
I did get the same error in an ".net Framework 4.8" project. It worked fine in an .net 6 project though.

**Reply by redemptions** [reply]
> Which framework do you guys use in the projects your testing with? I did get the same error in an ".net Framework 4.8" project. It worked fine in an .net 6 project though.

.net Framework 4.8, but the problem also disappears and comes back, which leads me to believe this was either server side or availability related.

**Reply by EmbedDutchDev** [reply]
After upgrading to 17.6.2 I have the same issue. Even the example questions like 'How does String.Substring() work?' give the Sorry, error fulfilling request. etc.
In my 17.5.x version the copilot chat did work as exected.  
The code pilot chat in my visual studio code still works

**Reply by EmbedDutchDev** [reply]
After right-click 'ask copilot' in the code is started working again... Not sure why, but it now works again, even the example questions in the chat window :-)

**Reply by dsarno** [reply]
The right click thing worked for me too.  Right clicking on a block of code in the main window, not the chat window.

**Comment by tomludd** [comment]
I too have this error in Visual Studio 2022 17.6.5.
To me the error only happens if I mark the first char on the first line the file, right click and select ask copilot.
I do not get the error if I use Ctrl + A to mark the whole document.

Seems like selecting the first character in the file results in a range starting on -1.
